pkg/controller/dnszone/dnszone_controller.go

```diff
@@ -215,10 +220,48 @@ func (r *ReconcileDNSZone) reconcileDNSProvider(actuator Actuator, dnsZone *hive
 	r.logger.Debug("Retrieving current state")
 	err := actuator.Refresh()
 	if err != nil {
+		if strings.Contains(err.Error(), "AccessDeniedException") {
```

Comment:
Can we check the error code instead of looking through the error string for "AccessDeniedException"? Cast the error to an `awserr.Error` and see what is in there. You may also be able to get the message that you want directly from the `awserr.Error` rather than doing regular expressions, too.